packages/block-editor/src/components/block-list/block-popover.js

```diff
@@ -235,6 +222,9 @@ function BlockPopover( {
 					__experimentalOnIndexChange={ ( index ) => {
 						initialToolbarItemIndexRef.current = index;
 					} }
+					// Resets the index whenever the active block changes so
+					// this is not persisted. See https://github.com/WordPress/gutenberg/pull/25760#issuecomment-717906169
+					key={ clientId }
```

Comment:
Remounting the component is better because the block element may change immediately from one to the other. Previously the block node would be undefined between client ID changes.